polyfactory/factories/base.py

```diff
@@ -487,11 +488,11 @@ def _create_generic_fn() -> Callable:
 
     @classmethod
     def create_factory(
-        cls,
-        model: type,
+        cls: type[F],
+        model: type[T],
```

Comment:
Some type-checkers issue an error when a type variable is only used once in a signature.

Basically, here it is the same as `type[Any]` which should be more correct.

Or is `T` used elsewhere?

Reply:
The `F` itself is generic since it's bound to `BaseFactory[Any]`. So by using `T`, the resulting type of the factory correctly resolves the types.

```python
from dataclasses import dataclass
from polyfactory.factories import DataclassFactory
from typing import reveal_type


@dataclass
class Foo:
    ...


FooFactory = DataclassFactory.create_factory(Foo)

reveal_type(FooFactory)  # type[DataclassFactory[Foo]]
```

Reply:
If `model` is typed as just `type`, then the resulting type for `FooFactory` is `type[DataclassFactory[Unknown]]`.

Reply:
Oh, I've missed that the class is generic on `T`, please ignore my comment